src/main/java/com/cdancy/bitbucket/rest/features/InsightsApi.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.features;
+
+import com.cdancy.bitbucket.rest.annotations.Documentation;
+import com.cdancy.bitbucket.rest.domain.common.RequestStatus;
+import com.cdancy.bitbucket.rest.domain.insights.Annotation;
+import com.cdancy.bitbucket.rest.domain.insights.AnnotationsResponse;
+import com.cdancy.bitbucket.rest.domain.insights.InsightReport;
+import com.cdancy.bitbucket.rest.domain.insights.InsightReportPage;
+import com.cdancy.bitbucket.rest.fallbacks.BitbucketFallbacks;
+import com.cdancy.bitbucket.rest.filters.BitbucketAuthenticationFilter;
+import com.cdancy.bitbucket.rest.options.CreateAnnotations;
+import com.cdancy.bitbucket.rest.options.CreateInsightReport;
+import com.cdancy.bitbucket.rest.parsers.RequestStatusParser;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Produces(MediaType.APPLICATION_JSON)
+@RequestFilters(BitbucketAuthenticationFilter.class)
+@Path("/rest/insights/{jclouds.api-version}")
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public interface InsightsApi {
+
+    @Named("insights:get-annotations")
+    @Documentation( {"https://docs.atlassian.com/bitbucket-server/rest/6.4.0/bitbucket-code-insights-rest.html#idp2"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/projects/{project}/repos/{repo}/commits/{commitId}/annotations")
+    @Fallback(BitbucketFallbacks.AnnotationsResponseOnError.class)
+    @GET
+    AnnotationsResponse listAnnotations(@PathParam("project") String project,
```

Comment:
:warning: Rather than using a lot of String arguments, consider using a container object for those values. (#QR14:62#)-PMD_UseObjectForClearerAPI